Source/CesiumRuntime/Private/CesiumGltfPrimitiveComponent.cpp

```diff
@@ -100,17 +54,82 @@ void UCesiumGltfPrimitiveComponent::BeginDestroy() {
 
     CesiumLifetime::destroy(pMesh);
   }
+}
+} // namespace
+void UCesiumGltfPrimitiveComponent::BeginDestroy() {
+  destroyCesiumPrimitive(this);
+  Super::BeginDestroy();
+}
 
+void UCesiumGltfInstancedComponent::BeginDestroy() {
+  destroyCesiumPrimitive(this);
   Super::BeginDestroy();
 }
 
 FBoxSphereBounds UCesiumGltfPrimitiveComponent::CalcBounds(
     const FTransform& LocalToWorld) const {
-  if (!this->boundingVolume) {
+  if (!getPrimitiveData()->boundingVolume) {
+    return Super::CalcBounds(LocalToWorld);
+  }
+
+  return std::visit(
+      CalcBoundsOperation{
+          LocalToWorld,
+          getPrimitiveData()->HighPrecisionNodeTransform},
+      *getPrimitiveData()->boundingVolume);
+}
+
+FBoxSphereBounds UCesiumGltfInstancedComponent::CalcBounds(
```

Comment:
This function and the one above are nearly identical. I'm not too fussed about it, but it might be nice to eliminate it since it's pretty easy to do. Something like this (not tested):

```cpp
namespace {

std::optional<FBoxSphereBounds> calcBounds(ICesiumPrimitive& primitive, const FTransform& LocalToWorld) {
  if (!primitive.boundingVolume) {
    return std::nullopt;
  }

  return std::visit(
      CalcBoundsOperation{
          LocalToWorld,
          primitive.HighPrecisionNodeTransform},
      *primitive.boundingVolume);
}

}

FBoxSphereBounds UCesiumGltfPrimitiveComponent::CalcBounds(
    const FTransform& LocalToWorld) const {
  std::optional<FBoxSphereBounds> maybeBounds = calcBounds(getPrimitiveData(), LocalToWorld);
  return maybeBounds ? *maybeBounds : Super::CalcBounds(LocalToWorld);
}

```